sky/clouds/service_catalog/common.py

```diff
@@ -325,3 +324,15 @@ def accelerator_in_region_or_zone_impl(
         return _accelerator_in_region(df, accelerator_name, acc_count, region)
     else:
         return _accelerator_in_zone(df, accelerator_name, acc_count, zone)
+
+
+# Images
+def get_image_id_from_tag_impl(df: pd.DataFrame, tag: str,
+                               region: Optional[str]) -> Optional[str]:
+    """Returns the image ID for the given tag and region."""
+    df = df[df['Tag'] == tag]
+    if region is not None:
+        df = df[df['Region'] == region]
+    assert len(
+        df) == 1, f'Found {len(df)} images for tag {tag} in region {region}'
+    return df['ImageId'].iloc[0]
```

Comment:
Document that this can be None, e.g., when a region doesn't have an image that matches the tag.